### DISCUSSION TITLE
whats difference between two controlnet one by lllyasviel and another by Mikubill

### DISCUSSION THREAD

**Discussion by AmitShashi** [lllyasviel/ControlNet#362] [General]
whats difference between two controlnet one by lllyasviel and another by Mikubill
which one i should use
which one is latest
do installing both of them at same time can cause any issue.
thanks in advance for replying.

**Comment by geroldmeisinger** [comment]
lllyasviel is the technology and architecture which enables to guide stable diffusion image generation, but you would have to commandline interfaces and python code to use it. Mikubill on the other hand is an extension for the Automatic1111 Stable Diffusion UI which provides convinience interface to use controlnet
